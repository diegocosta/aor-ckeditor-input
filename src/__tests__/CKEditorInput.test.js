import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import CKEditorInput from '../index';

const props = {
  meta: {},
};

describe('CKEditorInput', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CKEditorInput {...props} />, div);
  });

  it('renders activeClass correctly', () => {
    const localProps = {
      ...props,
      activeClass: 'aor-test',
    };

    const tree = renderer
      .create(<CKEditorInput {...localProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders error message correctly', () => {
    const localProps = {
      ...props,
      meta: {
        touched: true,
        error: 'Required',
      },
    };

    const tree = renderer
      .create(<CKEditorInput {...localProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
