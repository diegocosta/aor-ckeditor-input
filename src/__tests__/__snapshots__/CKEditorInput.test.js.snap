// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CKEditorInput renders activeClass correctly 1`] = `
<div
  className="aor-ckeditor-input"
>
  <div
    className="aor-test"
  />
</div>
`;

exports[`CKEditorInput renders error message correctly 1`] = `
<div
  className="aor-ckeditor-input"
>
  <div
    className=""
  />
  <div
    className="error"
  >
    Required
  </div>
</div>
`;
